### PR TITLE
Ignore llvm.instrprof.increment intrinsic

### PR DESCRIPTION
**Pull request by KornevNikita**
This is a workaround to avoid crash when translating
llvm.instrprof.increment which is not supported by SPIR-V and can not be
allowed by related option due to using llvm-spirv as a toolchain node.

### CODE REVIEW

**Comment by KornevNikita** [comment]
@AlexeySotkin @MrSidims @AlexeySachkov take a look please, do we need test here?

**Comment by MrSidims** [comment]
> do we need test here?

I'd say yes. See: test/llvm-intrinsics/

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -3189,6 +3189,8 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
   case Intrinsic::invariant_end:
   case Intrinsic::dbg_label:
   case Intrinsic::trap:
+  // llvm.instrprof.increment is not supported
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVWriter.cpp:3192]: Please rearrange the lines so comment about `llvm.trap`  appears just before `case Intrinsic::trap`.

File: lib/SPIRV/SPIRVWriter.cpp [hunk 2]
```diff
@@ -3189,6 +3189,8 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
...
+  case Intrinsic::instrprof_increment:
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVWriter.cpp:3193]: There are more `llvm.instrprof.*` intrinsics. I believe, if we want to add this one we should add the others too.

**Review by MrSidims — APPROVED** [review]

**Review by AlexeySotkin — APPROVED** [review]

**Comment by AlexeySotkin** [comment]
Probably it is not too late to squash them on the `master` branch if nobody objects :)

**Pull request merged**